### PR TITLE
Add more detail to the sync_summary dataset article

### PR DESCRIPTION
**Pull request by irrationalagent**
Looking to fill this out with some basic info. 

@linacambridge can you serve as sync expert reviewer here? I've noted a couple of places with your name because I wasn't entirely sure I was being accurate. Also: am I forgetting any engines?

### CODE REVIEW

**Comment by irrationalagent** [comment]
oops looks like i should have read the docs on the spellint, will fix that.

File: src/datasets/batch_view/sync_summary/reference.md [hunk 1]
```diff
@@ -7,26 +7,87 @@
...
+
+* `addons`
+* `addresses` mailing addresses e.g. for ecommerce.
+* `bookmarks`
+* `clients` non-user-facing list of the sync account's associated devices
+* `creditcards` this used to be nightly only but was recently removed entirely
+* `extension-storage` webextension-specific storage (ask Lina)?
+* `passwords`
+* `forms` saved values in webforms
+* `logins` deprecated?, use `passwords` (confirm with Lina)
```
Comment (linacambridge) [inline src/datasets/batch_view/sync_summary/reference.md]: Hmm, do we still have entries for `logins`? I thought we used `passwords` everywhere...

File: src/datasets/batch_view/sync_summary/reference.md [hunk 2]
```diff
@@ -7,26 +7,87 @@
...
+Existing engines (`engine_name` in `sync_flat_summary`) are listed below with brief descriptions in cases where their name isn't transparent.
+
+Note that not every device syncs each of these engines. They can be disabled individually and some are off by default.
+
+* `addons`
+* `addresses` mailing addresses e.g. for ecommerce.
+* `bookmarks`
+* `clients` non-user-facing list of the sync account's associated devices
+* `creditcards` this used to be nightly only but was recently removed entirely
+* `extension-storage` webextension-specific storage (ask Lina)?
```
Comment (linacambridge) [inline src/datasets/batch_view/sync_summary/reference.md]: Yep, this supports the `storage.sync` WebExtension API.

File: src/datasets/batch_view/sync_summary/reference.md [hunk 3]
```diff
@@ -7,26 +7,87 @@
...
+
+Existing engines (`engine_name` in `sync_flat_summary`) are listed below with brief descriptions in cases where their name isn't transparent.
+
+Note that not every device syncs each of these engines. They can be disabled individually and some are off by default.
+
+* `addons`
+* `addresses` mailing addresses e.g. for ecommerce.
```
Comment (linacambridge) [inline src/datasets/batch_view/sync_summary/reference.md]: Part of form autofill.

File: src/datasets/batch_view/sync_summary/reference.md [hunk 4]
```diff
@@ -7,26 +7,87 @@
...
+
+Existing engines (`engine_name` in `sync_flat_summary`) are listed below with brief descriptions in cases where their name isn't transparent.
+
+Note that not every device syncs each of these engines. They can be disabled individually and some are off by default.
+
```
Comment (linacambridge) [inline src/datasets/batch_view/sync_summary/reference.md]: `history`, too!

**Comment by irrationalagent** [comment]
@harterrt I think this is ready to go if you do (travis flagged a non-dead link as dead, not sure what's up there.)

**Pull request merged**